build-tools/packages/build-cli/src/codeCoverageAnalysisTool/getCoverageMetrics.ts

```diff
@@ -0,0 +1,148 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+/* eslint-disable no-await-in-loop */
+/* eslint-disable @typescript-eslint/dot-notation */
+/* eslint-disable @typescript-eslint/no-unsafe-argument */
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+
+import * as fs from "node:fs/promises";
+import * as path from "node:path";
+import globby from "globby";
+import JSZip from "jszip";
+import { Parser } from "xml2js";
+
+/**
+ * The type for the coverage report, containing the name of the package, line coverage and branch coverage
+ */
+export interface CoverageReport {
+	packageName: string;
+	lineCoverage: number;
+	branchCoverage: number;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any -- missing type for XML output
+const extractCoverageMetrics = (result: any): CoverageReport[] => {
```

Comment:
Is there any way to get better typing here? The comment indicates this is XML, but it doesn't seem to be parsed at any point.

Reply:
docs would also be helpful.